index.js

```diff
@@ -1,159 +1,98 @@
                 ]
             }
         ],
-        "at-rule-no-unknown": [
-            true,
+        'block-no-empty': true,
+        'color-hex-length': 'long',
+        'comment-no-empty': true,
+        'declaration-empty-line-before': 'never',
+        'font-weight-notation': 'numeric',
+        'keyframes-name-pattern': [
+            '^[a-z]+((\\d)|([A-Z0-9][a-z0-9]+))*([A-Z])*$',
             {
-                "ignoreAtRules": [
-                    "content",
-                    "extend",
-                    "include",
-                    "mixin",
-                    "container",
-                    "if",
-                    "for",
-                    "forward",
-                    "else",
-                    "elseif",
-                    "error",
-                    "each",
-                    "function",
-                    "return",
-                    "use",
-                    "while"
-                ]
-            }
+                message: (name) => `Expected keyframe name '${name}' to be camelCase`,
+            },
         ],
-        "at-rule-name-space-after": "always",
-        "function-comma-newline-before": "never-multi-line",
-        "function-name-case": [
-            "lower",
+        'no-descending-specificity': null,
+        'no-duplicate-selectors': true,
+        'no-empty-source': true,
+        'selector-class-pattern': [
+            '^[a-z]([-]?[a-z0-9]+)*(__[a-z0-9]([-]?[a-z0-9]+)*)?(--[a-z0-9]([-]?[a-z0-9]+)*)?$',
             {
-                "ignoreFunctions": [
-                    "/^[a-z][a-z-A-Z0-9]*$/"
-                ]
-            }
+                resolveNestedSelectors: true,
+                message: (selector) => `Expected class selector '${selector}' to match BEM CSS pattern (https://en.bem.info/methodology/css). Selector validation tool: https://regexr.com/3apms`,
+            },
         ],
-        "function-no-unknown": null, // https://github.com/stylelint-scss/stylelint-scss/issues/589
-        "block-closing-brace-newline-after": [
-            "always",
+        'selector-max-compound-selectors': [
+            3,
             {
-                "ignoreAtRules": [ "if", "else", "elseif" ]
+                severity: 'warning'
             }
         ],
-        "font-weight-notation": "numeric",
-        "string-quotes": "single",
-        "value-keyword-case": [
-            "lower",
+        'selector-max-id': 0,
+        'selector-max-type': [
+            2,
             {
-                camelCaseSvgKeywords: true,
+                severity: 'warning'
             }
         ],
-        "declaration-empty-line-before": "never",
-        "selector-attribute-quotes": "always",
-        "selector-class-pattern": "^[a-z][a-z-A-Z_0-9]*$",
-        "selector-max-compound-selectors": [
-            3,
+        'selector-max-combinators': [
+            2,
             {
-                "severity": "warning"
+                severity: 'warning'
```

Comment:
Not a fan of `warning` if there is a error and it make sense they should just add `stylelint-disable selector-max-compound-selectors` to that line.